README.md

```diff
@@ -1,97 +1,30 @@
+## Snap Plugin Library for Go
 
-This is a library for writing plugins for the [snap](https://github.com/intelsdi-x/snap) telemetry framework. The goal for this library is to make it super simple to write a plugin.
+This is a library for writing plugins in Go for the [Snap telemetry framework](https://github.com/intelsdi-x/snap). 
 
-# Writing a plugin
+Snap has three different plugin types and for instructions on how to write a plugin check out the [collector](examples/collector/README.md), [processor](examples/processor/README.md), and [publisher](examples/publisher/README.md) plugin docs.
 
-In order to write a plugin for snap, it is necessary to define some methods to satisfy the appropriate interfaces. The interface is slightly different depending on what type (collector, processor, or publisher) of plugin is being written. The interfaces are:
+Before writing a Snap plugin:
 
+* See if one already exists in the [Plugin Catalog](https://github.com/intelsdi-x/snap/blob/master/docs/PLUGIN_CATALOG.md) 
+* See if someone mentioned it in the [plugin wishlist](https://github.com/intelsdi-x/snap/labels/plugin-wishlist)
 
-```go
-type Plugin interface {
-	GetConfigPolicy() (ConfigPolicy, error)
-}
+If you do decide to write a plugin, open a new issue following the plugin [wishlist guidelines](https://github.com/intelsdi-x/snap/blob/master/docs/PLUGIN_CATALOG.md#wish-list) and let us know you are working on one!
 
-// Collector is a plugin which is the source of new data in the Snap pipeline.
-type Collector interface {
-	Plugin
 
-	GetMetricTypes(Config) ([]Metric, error)
-	CollectMetrics([]Metric) ([]Metric, error)
-}
+## Brief Overview of Snap Architecture
 
-// Processor is a plugin which filters, agregates, or decorates data in the
-// Snap pipeline.
-type Processor interface {
-	Plugin
+Snap is an open and modular telemetry framework designed to simplify the collection, processing and publishing of data through a single HTTP based API. Plugins provide the functionality of collection, processing and publishing and can be loaded/unloaded, upgraded and swapped without requiring a restart of the Snap daemon.
 
-	Process([]Metric, Config) ([]Metric, error)
-}
+A Snap plugin is a program that responds to a set of well defined [gRPC](http://www.grpc.io/) services with parameters and return types specified as protocol buffer messages (see [plugin.proto](https://github.com/intelsdi-x/snap/blob/master/control/plugin/rpc/plugin.proto)). The Snap daemon handshakes with the plugin over stdout and then communicates over gRPC.
 
-// Publisher is a sink in the Snap pipeline.  It publishes data into another
-// System, completing a Workflow path.
-type Publisher interface {
-	Plugin
 
-	Publish([]Metric, Config) error
-}
-```
+## Snap Plugin Go Library Implementation Files
+You will find all the library packages in the [v1/plugin folder](v1/plugin).
+The [rpc folder](v1/plugin/rpc) contents are auto-generated and should not be modified by anyone.
```

Comment:
originally I was trying to provide info on what's included in the repo, but with all other updates this is probably unnecessary, went ahead and took this section out.